crates/cxx-qt-gen/src/writer/rust/mod.rs

```diff
@@ -58,6 +58,15 @@ fn cxx_qt_common_blocks(qobject: &GeneratedRustQObject) -> Vec<TokenStream> {
     let create_rs_ident = mangle("create_rs", rust_struct_ident);
 
     vec![
+        quote! {
+            impl core::ops::Deref for #cpp_struct_ident {
```

Comment:
note we can't quite do `DerefMut` as the `Pin<&mut T>` makes that not work as the C++ type is not `Unpin`.